### PR TITLE
[New Model] Add Falcon Perception support

### PR DESCRIPTION
**Pull request by avirajBevli**
## Purpose
Add vLLM-Omni support for Falcon Perception, which is a vision-language model for open-vocabulary instance segmentation
https://github.com/tiiuae/Falcon-Perception
https://huggingface.co/tiiuae/Falcon-Perception

This model follows a **2 stage pipeline**:
- **Stage0:** thinker stage (AR LLM): generates the `<coord>`, `<size>`, `<seg>` tokens autoregressively for each object instance. Object coordinates and sizes are decoded from the hidden states corresponding to the the sampled `<coord>`,` <size>` tokens. Their embeddings are then injected back into the next autoregressive decoding step.
- **Stage1:** segmentation stage : takes in the patch features, hidden states of <seg> tokens of each object instance as input. And as output, generates one segmentation mask per detected object instance

## Test Plan
**vLLM Version:** `0.26.0`

## Test Result
Examples of output segmentation masks overlayed on top of the original image.

- Query = birds
- Input image
image

- Output segmentation mask overlayed on top of the original image
image

### L2 test : pass
```
python -m pytest -s -v \
  tests/e2e/offline_inference/test_falcon_perception.py \
  -m "core_model and cuda" \
  --run-level core_model
```
image

### L3 test result: pass
```
VLLM_USE_FLASHINFER_SAMPLER=0 \
python -m pytest -s -v \
  tests/e2e/offline_inference/test_falcon_perception.py \
  -m "advanced_model and cuda" \
  --run-level advanced_model
```
image

## Out of scope
Support for Falcon perception detection-only model is not added here. That is left for a future PR

**BEFORE SUBMITTING:** read [CONTRIBUTING.md](https://github.com/vllm-project/vllm-omni/blob/main/CONTRIBUTING.md) and run the [precheck-pr skill](https://github.com/vllm-project/vllm-omni/blob/main/.claude/skills/precheck-pr/SKILL.md) with the code agent for a self-check against project conventions.
(anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

**Comment by avirajBevli** [comment]
@hsliuustc0106  
Request to have a look at the PR when convenient. 
Request adding ready label so the 1 pending check (buildkite) can run? 

Note that the other 5 checks ran successfully.

File: vllm_omni/model_executor/models/falcon_perception/falcon_perception_segmentation.py [hunk 1]
```diff
@@ -0,0 +1,600 @@
...
+        raise ValueError(f"binary masks must have shape (N, H, W), got {tuple(binary_masks.shape)}")
+
+    count, height, width = binary_masks.shape
+    if count <= 1:
+        return torch.arange(count, device=binary_masks.device, dtype=torch.long)
+
+    scale = min(1.0, nms_max_side / max(height, width))
+    target_h = max(1, int(round(height * scale)))
+    target_w = max(1, int(round(width * scale)))
+    masks = binary_masks.float()
```
Comment (lishunyang12) [inline vllm_omni/model_executor/models/falcon_perception/falcon_perception_segmentation.py]: Downsample masks in chunks; this full-resolution float copy defeats the memory bound.

File: vllm_omni/model_executor/models/falcon_perception/falcon_perception_thinker.py [hunk 2]
```diff
@@ -0,0 +1,1030 @@
...
+
+        Same contract as the in-tree talker stages: the runner writes the
+        returned dict into its per-request intermediate buffer, so nothing is
+        kept on the module and continuous batching stays safe.
+        """
+        # Runner-called, once per decode step, outside the compiled model.
+        with nvtx_range("fp/s0/postprocess"):
+            if hidden_states.numel() == 0:
+                return {}
+            return {"hidden_states": {"last": hidden_states[-1].detach().contiguous()}}
```
Comment (lishunyang12) [inline vllm_omni/model_executor/models/falcon_perception/falcon_perception_thinker.py:717]: Mark hidden_states.last GPU-resident; the runner currently copies it to CPU every token.

File: examples/offline_inference/falcon_perception/batch_inference.py [hunk 3]
```diff
@@ -0,0 +1,437 @@
...
+    (48, 209, 88),
+]
+
+
+def build_prompt(query: str) -> str:
+    """The exact string the model expects. Deviating here silently degrades output."""
+    return f"<|image|>Segment these expressions in the image:<|start_of_query|>{query}<|REF_SEG|>"
+
+
+def overlay(image: Image.Image, masks: np.ndarray) -> Image.Image:
```
Comment (lishunyang12) [inline examples/offline_inference/falcon_perception/batch_inference.py]: Reuse end2end.py's prompt and overlay helpers instead of duplicating them here.

File: vllm_omni/model_executor/models/falcon_perception/falcon_perception_segmentation.py (hunk 1 above)
Comment (avirajBevli) [reply vllm_omni/model_executor/models/falcon_perception/falcon_perception_segmentation.py]: Makes sense. Fixed this

File: vllm_omni/model_executor/models/falcon_perception/falcon_perception_thinker.py (hunk 2 above)
Comment (avirajBevli) [reply vllm_omni/model_executor/models/falcon_perception/falcon_perception_thinker.py:717]: Makes sense to avoid useless gpu -> cpu -> gpu transfer from postprocess of current step to preprocess of the next step. Fixed this

File: examples/offline_inference/falcon_perception/batch_inference.py (hunk 3 above)
Comment (avirajBevli) [reply examples/offline_inference/falcon_perception/batch_inference.py]: Created a _common file from which both import. hence avoid duplicate
